src/vellum/workflows/state/context.py

```diff
@@ -145,7 +147,32 @@ def resolve_workflow_deployment(self, deployment_name: str, release_tag: str) ->
         Returns:
             BaseWorkflow instance if found, None otherwise
         """
-        return None
+        if not self.generated_files:
+            return None
+
+        expected_hash = str(uuid4_from_hash(f"{deployment_name}|{release_tag}")).replace("-", "_")
+        expected_prefix = f"vellum_workflow_deployment_{expected_hash}"
+
+        workflow_file_key = f"{expected_prefix}/workflow.py"
+        if workflow_file_key not in self.generated_files:
+            return None
+
+        from vellum_ee.workflows.server.virtual_file_loader import VirtualFileFinder
+
+        finder = VirtualFileFinder(self.generated_files, expected_prefix)
+        sys.meta_path.append(finder)
```

Comment:
we shouldn't need another `VirtualFileFinder`.

Instead, we should be able to do a relative import from the root namespace, eg `BaseWorkflow.load_from_module(f".{expected_prefix}")`

Reply:
If the test fails after trying this, let it stay failed in CI so I could take a look